test/system/200-pod.bats

```diff
@@ -478,27 +483,57 @@ spec:
+    assert "$output" != "" "losetup: output"
+
+    lodevice=$(awk "\$1 == \"$lofile\" { print \$2 }" <<<"$output")
+    lomajmin=$(awk "\$1 == \"$lofile\" { print \$3 }" <<<"$output")
+
+    is "$lodevice" ".\+" "Could not determine device for $lofile"
+    is "$lomajmin" ".\+" "Could not determine major/minor for $lofile"
+
+    run grep -w bfq /sys/block/$(basename ${lodevice})/queue/scheduler
+    if [ $status -ne 0 ]; then
+        skip "BFQ scheduler is not supported on the system"
+    fi
+    echo bfq > /sys/block/$(basename ${lodevice})/queue/scheduler
+
+    expected_limits="
+cpu.max         | 500000 100000
+memory.max      | 5242880
+memory.swap.max | 1068498944
+io.max          | $lomajmin rbps=1048576 wbps=1048576 riops=max wiops=max
+"
+
     local name1="resources1"
-    run_podman --cgroup-manager=systemd pod create --name=$name1 --cpus=5 --memory=10m
+    run_podman --cgroup-manager=systemd pod create --name=$name1 --cpus=5 --memory=5m --memory-swap=1g --cpu-shares=1000 --cpuset-cpus=0 --cpuset-mems=0 --device-read-bps=${lodevice}:1mb --device-write-bps=${lodevice}:1mb --blkio-weight-device=${lodevice}:123 --blkio-weight=50
     run_podman --cgroup-manager=systemd pod start $name1
     run_podman pod inspect --format '{{.CgroupPath}}' $name1
-    local path1="$output"
-    local actual1=$(< /sys/fs/cgroup/$path1/cpu.max)
-    is "$actual1" "500000 100000" "resource limits set properly"
-    local actual2=$(< /sys/fs/cgroup/$path1/memory.max)
-    is "$actual2" "10485760" "resource limits set properly"
+    local cgroup_path="$output"
+
+    while read unit expect; do
+        local actual=$(< /sys/fs/cgroup/$cgroup_path/$unit)
+        is "$actual" "$expect" "resource limit: $unit"
+    done < <(parse_table "$expected_limits")
     run_podman pod --cgroup-manager=systemd rm -f $name1
 
     local name2="resources2"
-    run_podman --cgroup-manager=cgroupfs pod create --cpus=5 --memory=10m --name=$name2
+    run_podman --cgroup-manager=cgroupfs pod create --cpus=5 --memory=5m --name=$name2 --memory-swap=1g --cpu-shares=1000 --cpuset-cpus=0 --cpuset-mems=0 --device-read-bps=${lodevice}:1mb --device-write-bps=${lodevice}:1mb --blkio-weight-device=${lodevice}:123 --blkio-weight=50
     run_podman --cgroup-manager=cgroupfs pod start $name2
     run_podman pod inspect --format '{{.CgroupPath}}' $name2
-    local path2="$output"
-    local actual2=$(< /sys/fs/cgroup/$path2/cpu.max)
-    is "$actual2" "500000 100000" "resource limits set properly"
-    local actual2=$(< /sys/fs/cgroup/$path2/memory.max)
-    is "$actual2" "10485760" "resource limits set properly"
-    run_podman --cgroup-manager=cgroupfs pod rm $name2
+    local cgroup_path="$output"
+
+    while read unit expect; do
+        local actual=$(< /sys/fs/cgroup/$cgroup_path/$unit)
+        is "$actual" "$expect" "resource limit: $unit"
+    done < <(parse_table "$expected_limits")
+    run_podman pod --cgroup-manager=cgroupfs rm -f $name2
```

Comment:
Thank you, this is all much more readable now, but the problem with making things readable is that one discovers new problems :-)

As best I can tell, lines 527-536 are 100% identical to 515-524, with the only difference being `--cgroup-manager`. (And `--name`, which is irrelevant). Could you rewrite this as
```bash
    for cgm in systemd cgroupfs; do
        name=resources-$cgm
        run_podman --cgroup-manager=$cgm ...
        ....
        is "$actual" "$expect" "resource limit under $cgm: $unit"
```